OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationPermissionController.kt

```diff
@@ -42,15 +42,15 @@ object NotificationPermissionController : PermissionsActivity.PermissionCallback
         PermissionsActivity.registerAsCallback(PERMISSION_TYPE, this)
     }
 
-    @ChecksSdkIntAtLeast(api = 33)
-    val supportsNativePrompt =
-        Build.VERSION.SDK_INT > 32 &&
-            OSUtils.getTargetSdkVersion(OneSignal.appContext) > 32
-
     fun prompt(
         fallbackToSettings: Boolean,
         callback: OneSignal.PromptForPushNotificationPermissionResponseHandler?
     ) {
+
+        val supportsNativePrompt =
```

Comment:
This make this code imperative instead of declarative which makes this harder to read.

I would recommend converting this a [Kotlin Lazy property](https://kotlinlang.org/docs/delegated-properties.html#lazy-properties)